microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,514 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re, time
+
+from assertpy import assert_that
+
+from lisa import (
+    Node,
+    TestSuite,
+    TestSuiteMetadata,
+    TestCaseMetadata,
+    Environment
+)
+from lisa.base_tools.cat import Cat
+from lisa.operating_system import SLES, CentOs, Redhat
+from lisa.testsuite import TestCaseMetadata
+from lisa.tools.echo import Echo
+from lisa.util import SkippedException
+from lisa.util.logger import Logger
+
+def check_vm_feature_support_status(min_support_version: str, node: Node) -> bool:
+    if not min_support_version:
+        return False
+    
+    kernel_array = node.execute(
+        "uname -r | awk -F '[.-]' '{print $1,$2,$3,$4}'"
+    ).stdout.split()
+    support_kernel_array = re.split('\.|\-', min_support_version)
+    for indx in range(4):
+        if kernel_array[indx] > support_kernel_array[indx]:
+            return True
+        elif kernel_array[indx] < support_kernel_array[indx]:
+            return False
+    return True
+
+
+@TestSuiteMetadata(
+    area="power",
+    category="functional",
+    description="""
+    """
+)
+class power(TestSuite):
+    HB_CUSTOM_KERNEL_URL = "git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git"
+    HB_CUSTOM_KERNEL_BRANCH = "stable"
+    HB_URL = ""
+
+    @TestCaseMetadata(
+        description="""
+        This test can be performed in Azure and Hyper-V both. But this script only covers Azure.
+        1. Prepare swap space for hibernation
+        2. Compile a new kernel (optional)
+        3. Update the grub.cfg with resume=UUID=xxxx where is from blkid swap disk
+        4. Set RTC to future time or past time
+        5. Hibernate the VM
+        5. Resume the VM
+        6. Read RTC timestamp and compare to the original value
+        """,
+        priority=3
+    )
+    def power_clock_sync_hibernate(self, environment: Environment, node: Node, log: Logger) -> None:
+        max_kernel_compile_min = 90
+        
+        # Prepare the swap space in the target VM
+        resource_group = 
+        vm_name = 
+        location = 
+        storage_type = "StandardSSD_LRS"
```

Comment:
If the disktype is required, use the disk feature. But if there is no special reason, not to limit the disk type, it makes test case cannot run under other conditions.